projecttacoma/fqm-testify
# Test case Measure Report Import

# Summary
Importing a patient bundle with a cqfm test case MeasureReport will now update that patient's desired population.

## New Behavior
When importing a patient bundle or a .zip file of patient bundles with cqfm test case Measure Report resources, the desired population of the patient test case will reflect the populations scored in the cqfm test case measure report. cqfm test case MeasureReports will no longer be viewable in the app when imported as part of a patient bundle.

## Code Changes
 - Updated bundleToTestCase function to take new `populationGroups` parameter, detect presence of test case MeasureReport, parse it into desired populations when appropriate, and exclude it from resource creation.
 - Created new isTestCaseMeasureReport helper function
 - Converted existing `getMeasurePopulations` function to `getMeasurePopulationsForSelection`
 - Created new `getMeasurePopulations` function that returns just the codes from the populations

# Testing Guidance
- `npm run check`
- `npm run dev`
- upload `EXM130` bundle
- Try uploading the test cases from the attached file. Unzip the first .zip file to reveal individual test cases and a nested .zip file. Uploading the initial .zip as is will throw an error
- The test cases are named after the expected outcome for each upload (ex. Nancy McNumerator desires to be in the numerator)
- `Wannabe Numex` and `Multiple Test Case Mrs` should both throw comprehensive errors on attempted upload
[test-cases.zip](https://github.com/projecttacoma/fqm-testify/files/10107666/test-cases.zip)
